Fix false negative in `Lint/Void` with initialize and setter methods

pocke opened the pull request:
In `initialize` and setter methods, returned value is ignored(See #4153).

```ruby
def initialize
  42 # This value is ignored.
end
```

However, `Lint/Void` cop doesn't add an offense for the above code.

This change makes to add an offense for the code.


-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

Drenmi commented:
Hi, @pocke!

There's work started on this in #4201. Maybe you two can collaborate? 🙂

bbatsov commented on spec/rubocop/cop/lint/void_spec.rb:
```diff
@@ -65,6 +65,38 @@
+  end
+
+  it 'registers two offenses for void literals in an initialize method' do
+    expect_offense(<<-END.strip_indent)
```
Use `RUBY` everywhere consistently. We really need a cop to warn for usages of `END`.

bbatsov commented on lib/rubocop/cop/lint/void.rb:
```diff
@@ -102,6 +104,17 @@ def check_for_defined(node)
           add_offense(node, :expression, format(DEFINED_MSG, node.source))
         end
+
+        # `initialize` and setter methods is not return value of last
```
do not return

bbatsov commented on lib/rubocop/cop/lint/void.rb:
```diff
@@ -102,6 +104,17 @@ def check_for_defined(node)
+          parent = node.parent
+          return false unless parent
+          method = parent.children.first
+          parent.def_type? &&
```
@Drenmi I see a `def` node extension potential here. :-)

Drenmi replied:
Yes. Both `def` and `defs` have been on my list for some time. They're probably next. 🙂

pocke commented:
@bbatsov Thanks for your reviewing. I updated the commit, and rebased.

@Drenmi 

> There's work started on this in #4201. Maybe you two can collaborate? 

Hmm, I can collaborate on the pull-request if @harold-s is busy.

bbatsov commented:
> Hmm, I can collaborate on the pull-request if @harold-s is busy.

That'd be best - he hasn't touched this in months and I have doubts he'll pick it up any time soon.
